### PR TITLE
refactor: inject the state into sub componnets of the main component …

### PR DESCRIPTION
**Pull request by Keith-CY**
…and remove the state from hoc

### CODE REVIEW

File: packages/neuron-ui/src/stories/Overview.stories.tsx [hunk 1]
```diff
@@ -61,14 +61,11 @@ const states = {
...
 
-const OverviewWithRouteProps = (props: StateWithDispatch) => (
-  <Route path="/" render={routeProps => <Overview {...routeProps} {...props} />} />
-)
-
 const stories = storiesOf(`Overview`, module).addDecorator(StoryRouter())
 
 Object.entries(states).forEach(([title, props]) => {
-  stories.add(title, () => <OverviewWithRouteProps {...props} />)
+  console.info(props)
```
Comment (classicalliu) [inline packages/neuron-ui/src/stories/Overview.stories.tsx]: Is this log expected ?

File: packages/neuron-ui/src/stories/Overview.stories.tsx [hunk 2]
```diff
@@ -120,5 +117,6 @@ stories.addDecorator(withKnobs).add('With knobs', () => {
       ],
     },
   }
-  return <OverviewWithRouteProps {...props} />
+  console.info(props)
```
Comment (classicalliu) [inline packages/neuron-ui/src/stories/Overview.stories.tsx]: Also here.

File: packages/neuron-ui/src/stories/Overview.stories.tsx (hunk 1 above)
Comment (Keith-CY) [reply packages/neuron-ui/src/stories/Overview.stories.tsx]: Yes, it's used to print the expected state of the component in the storybook. Due to the `useContext` hook used in these components, the states cannot be observed directly.

Comment (classicalliu) [reply packages/neuron-ui/src/stories/Overview.stories.tsx]: ok

Comment (Keith-CY) [reply packages/neuron-ui/src/stories/Overview.stories.tsx]: BTW, all the codes in `/stories` are just for developing and testing.

**Pull request merged**